### PR TITLE
[#471] Remove unused fields from UpdateData

### LINKED ISSUE #471: Remove unused fields from UpdateData
## Background

@cleverca22 confirmed that only one of the four fields of `UpdateData` are used, e.g.:

```haskell
(SystemTag {getSystemTag = "linux64"},
 UpdateData {
   udAppDiffHash = AbstractHash > 03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314,
   udPkgHash = AbstractHash 86bdd254fe6a5e9c0da81dbf591f2f63b2f688c5ec147c6ace474dd36aa74da2,
   udUpdaterHash = AbstractHash 03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314,
   udMetadataHash = AbstractHash 03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314})
```
`udAppDiffHash`, `udUpdaterHash`, and `udMetadataHash` are identical. Moreover, `03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314` seems to the hash of an empty string.

## Proposed Solution

- Keep the same binary format, but retain the parts we care about.
- During decoding, assert that the ignored fields are set to `03170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c111314`

### PR DESCRIPTION
**Pull request by ruhatch**
Closes #471

### CODE REVIEW

File: src/Cardano/Chain/Update/Data.hs [hunk 1]
```diff
@@ -17,57 +17,36 @@ import Cardano.Binary
...
-  , udUpdaterHash  :: !(Hash Raw)
-  -- ^ Hash if update application which can be used to install this update
-  --   (relevant only when updater is used, not package).
-  , udMetadataHash :: !(Hash Raw)
-  -- ^ Hash of metadata relevant to this update. It is raw hash, because
-  --   metadata can include image or something (maybe). Anyway, we can always
-  --   use `unsafeHash`.
+-- | The update data is the hash of the installer of the new application
+newtype UpdateData = UpdateData
+  { unUpdateData :: Hash Raw
```
Comment (dnadales) [inline src/Cardano/Chain/Update/Data.hs]: Would it make sense to rename this to something like `InstallerHash`? Or do we want to leave open the possibility of adding more fields to `UpdateData`?

**Review by dnadales — APPROVED** [review]
:tada:

File: src/Cardano/Chain/Update/Data.hs (hunk 1 above)
Comment (ruhatch) [reply src/Cardano/Chain/Update/Data.hs]: I'm sure we won't, so yeah I'll do that

**Pull request merged**